README.md

```diff
@@ -130,21 +185,96 @@ impl FutureService for HelloServer {
 }
 
 fn main() {
-    let mut reactor = reactor::Core::new().unwrap();
-    let (handle, server) = HelloServer.listen("localhost:10000".first_socket_addr(),
-                                  &reactor.handle(),
-                                  server::Options::default())
-                          .unwrap();
-    reactor.handle().spawn(server);
-    let options = client::Options::default().handle(reactor.handle());
-    reactor.run(FutureClient::connect(handle.addr(), options)
+    let matches = clap::App::new("hello sync client")
+        .arg(clap::Arg::with_name("port").required(true))
+        .get_matches();
+    let port: u16 = matches.value_of("port").unwrap().parse().unwrap();
+    let mut reactor = tokio_core::reactor::Core::new().unwrap();
+    let (handle, server) = HelloServer.listen(format!("localhost:{}", port).first_socket_addr(),
+                &reactor.handle(),
+                Options::default())
+        .unwrap();
+    println!("Listening on {}", handle.addr());
+    reactor.run(server).unwrap();
+}
+```
+
+In this example, `hello` returns `Result`, but in more complex servers, you
+might talk to another server or do something using IO. In those cases, the
+`HelloFut` type would be something more complex. An example of such a server
+can be found in `examples/two_servers.rs`. In cases where RPC handlers are very
+fast and don't block, implementing `FutureService` is a good choice, since
+there's no threading overhead. However, blocking in an RPC method
+implementation would cause the reactor core running the server to grind to a
+halt, which would prevent new RPCs from being served. Blocking operations can
+still be used, but their execution must be delegated to a thread. A common way
+to do that is to use `futures-cpupool`.
+
+The `listen` method returns the server handle and the server future. Running
+this future on a reactor core causes the server to serve incoming requests.
+Having the future itself returned gives you the flexibility to run the server
+either using a reactor handle (`handle.spawn()`), or by running it on the
+reactor directly. Running the server future on a handle is useful if you might
+run multiple servers on the same reactor. After calling `listen` we need to run
+the server on a reactor core. The returned future will not resovle unless
+`handle.shutdown().shutdown()` is called. In this particular code snippet, it
```

Comment:
I'm afraid about this bitrotting because we're going to change this soon. I guess there's not a good options at the moment. Maybe just say "unless the server is shut down" and leave the rest as an exercise to the reader.